File: order-service/src/main/java/com/example/order/controller/OrderController.java
package com.example.order.controller;

import com.example.order.dto.OrderWithUserResponse;
import com.example.order.entity.Order;
import com.example.order.service.OrderService;
import com.example.order.service.OrderService.CreateOrderRequest;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@Slf4j
@RestController
@RequestMapping("/orders")
@RequiredArgsConstructor
public class OrderController {

    private final OrderService orderService;

    @PostMapping
    public ResponseEntity<Order> createOrder(@RequestBody CreateOrderRequest request) {
        log.info("[Order Controller] 주문 생성 API 호출됨");
        Order order = orderService.createOrder(request);
        log.info("[Order Controller] 주문 생성 완료 - orderId: {}", order.getId());
        return ResponseEntity.status(HttpStatus.CREATED).body(order);
    }

    @GetMapping
    public ResponseEntity<?> getOrders(@RequestParam(required = false) Long userId) {
        if (userId != null) {
            // userId가 있으면 Order + User 정보 함께 반환
            log.info("[Order Controller] userId로 주문 조회 (User 정보 포함) - userId: {}", userId);
            List<OrderWithUserResponse> ordersWithUser = orderService.getOrdersWithUserInfo(userId);
            return ResponseEntity.ok(ordersWithUser);
        } else {
            // userId가 없으면 전체 주문 조회 (Order만)
            log.info("[Order Controller] 전체 주문 조회");
            List<Order> orders = orderService.getAllOrders();
            return ResponseEntity.ok(orders);
        }
    }
}

File: src/main/kotlin/com/example/monolith/order/OrderController.kt
package com.example.monolith.order

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/orders")
class OrderController(
    private val orderService: OrderService
) {

    @PostMapping
    fun createOrder(@RequestBody request: CreateOrderRequest): ResponseEntity<Order> {
        val order = orderService.createOrder(request)
        return ResponseEntity.status(HttpStatus.CREATED).body(order)
    }

    @GetMapping
    fun getAllOrders(): ResponseEntity<List<Order>> {
        val orders = orderService.getAllOrders()
        return ResponseEntity.ok(orders)
    }

    @GetMapping("/{id}")
    fun getOrderById(@PathVariable id: Long): ResponseEntity<Order> {
        val order = orderService.getOrderById(id)
        return ResponseEntity.ok(order)
    }

    @GetMapping("/user/{userId}")
    fun getOrdersByUserId(@PathVariable userId: Long): ResponseEntity<List<Order>> {
        val orders = orderService.getOrdersByUserId(userId)
        return ResponseEntity.ok(orders)
    }

    @PatchMapping("/{id}/status")
    fun updateOrderStatus(
        @PathVariable id: Long,
        @RequestBody request: UpdateOrderStatusRequest
    ): ResponseEntity<Order> {
        val order = orderService.updateOrderStatus(id, request.status)
        return ResponseEntity.ok(order)
    }
}

data class UpdateOrderStatusRequest(
    val status: OrderStatus
)